FlowCrypt/Functionality/Services/Compose Message Service/ComposeMessageService.swift

```diff
@@ -122,36 +110,57 @@ final class ComposeMessageService {
             subject: subject,
             replyToMimeMsg: replyToMimeMsg,
             atts: sendableAttachments,
-            pubKeys: [myPubKey] + allRecipientPubs,
-            signingPrv: signingPrv
+            pubKeys: [myPubKey] + validPubKeys,
+            signingPrv: signingPrv,
+            password: contextToSend.password
         )
     }
 
-    private func getPubKeys(for recipients: [ComposeMessageRecipient]) async throws -> [String] {
+    private func getRecipientKeys(for recipients: [ComposeMessageRecipient]) async throws -> [RecipientWithSortedPubKeys] {
         var recipientsWithKeys: [RecipientWithSortedPubKeys] = []
         for recipient in recipients {
             let armoredPubkeys = contactsService.retrievePubKeys(for: recipient.email).joined(separator: "\n")
             let parsed = try await self.core.parseKeys(armoredOrBinary: armoredPubkeys.data())
             recipientsWithKeys.append(RecipientWithSortedPubKeys(email: recipient.email, keyDetails: parsed.keyDetails))
         }
-        return try validate(recipients: recipientsWithKeys)
+        return recipientsWithKeys
     }
 
-    private func validate(recipients: [RecipientWithSortedPubKeys]) throws -> [String] {
+    private func validate(recipients: [RecipientWithSortedPubKeys], withMessagePassword: Bool) throws -> [String] {
         func contains(keyState: PubKeyState) -> Bool {
             recipients.first(where: { $0.keyState == keyState }) != nil
         }
+
+        func hasRecipientsWithoutPubKey(withPasswordSupport: Bool) -> Bool {
+            recipients
+                .filter { $0.keyState == .empty }
+                .first(where: {
+                    guard let domain = $0.email.recipientDomain else { return !withPasswordSupport }
+                    let supportsPassword = domainsWithPasswordSupport.contains(domain)
+                    return withPasswordSupport == supportsPassword
+                }) != nil
```

Comment:
Same here - can be removed. We only care that recipient has no pubkey, and sender is on `flowcrypt.com` domain - then we allow the functionality.